src/mixins/NextcloudMixin.js

```diff
@@ -0,0 +1,208 @@
+            + ' The server-info API may return this error for incorrect user/password.',
+          );
+          break;
+        case 993:
+        case 997:
+        case 998:
+          this.validCredentials = false;
+          this.error(
+            'The provided app-password is not permitted to access the requested resource or it has'
+            + ' been revoked, or the username/password combination is incorrect',
+          );
+          break;
+        default:
+          this.validCredentials = true;
+          if (!this.allowedStatuscodes().includes(meta.statuscode)) {
+            this.error('Unexpected response');
+          }
+          break;
+      }
+      return data;
+    },
+    /* Process the capabilities endpoint if {capabilitiesTtl} has expired */
+    loadCapabilities() {
+      if ((new Date().getTime()) - this.capabilitiesLastUpdated > this.capabilitiesTtl) {
+        return this.makeRequest(this.endpoint('capabilities'), this.headers)
+          .then(this.processCapabilities);
+      }
+      return Promise.resolve();
+    },
+    /* Update the sate based on the capabilites response */
+    processCapabilities(capResponse) {
+      const ocdata = this.validateResponse(capResponse);
+      const capNotif = ocdata.capabilities?.notifications?.['ocs-endpoints'];
+      this.branding = ocdata.capabilities?.theming;
+      this.capabilities.notifications.enabled = !!(capNotif?.length);
+      this.capabilities.notifications.features = capNotif || [];
+      this.capabilities.userStatus = !!(ocdata.capabilities?.user_status?.enabled);
+      this.version.string = ocdata.version?.string;
+      this.version.edition = ocdata.version?.edition;
+      this.capabilitiesLastUpdated = new Date().getTime();
+    },
+    /* Shared template helpers */
+    getTimeAgo(time) {
+      return getTimeAgo(time);
+    },
+    formatDateTime(time) {
+      return timestampToDateTime(time);
+    },
+    /* Add additional formatting to {MiscHelpers.convertBytes()} */
+    convertBytes(bytes, decimals = 2, formatHtml = true) {
+      const formatted = convertBytes(bytes, decimals).toString();
+      if (!formatHtml) return formatted;
+      const m = formatted.match(/(-?\d+)((\.\d+)?\s(([KMGTPEZY]B|Bytes)))/);
+      return `${m[1]}<span class="decimals">${m[2]}</span>`;
+    },
+    /* Add additional formatting to {MiscHelpers.formatNumber()} */
+    formatNumber(number, decimals = 1, formatHtml = true) {
+      const formatted = formatNumber(number, decimals).toString();
+      if (!formatHtml) return formatted;
+      const m = formatted.match(/(\d+)((\.\d+)?([KMBT]?))/);
+      return `${m[1]}<span class="decimals">${m[2]}</span>`;
+    },
+    /* Format a number as percentage value */
+    formatPercent(number, decimals = 2) {
+      const n = parseFloat(number).toFixed(decimals).split('.');
+      const d = n.length > 1 ? `.${n[1]}` : '';
+      return `${n[0]}<span class="decimals">${d}%</span>`;
+    },
```

Comment:
The source of the injected HTML In all cases [Sonar marked](https://sonarcloud.io/project/security_hotspots?id=Dashy&pullRequest=740&sinceLeakPeriod=true) as vulnerable to XSS due to using `v-html` is trusted (above methods), and the input, though could be forged by a malicious actor, is sanitised by `parseFloat()` so I wouldn't consider this serious.

On the other hand, the sole purpose of this injection is aesthetic and has no functional significance so it can be removed.

Reply:
Yeah, given where and how this is used, I feel that in this case it's okay